### PR TITLE
[CARBONDATA-410] added test case for file system

### PR DESCRIPTION
**Pull request by anubhav100**

### CODE REVIEW

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    private static String fileName;
+    private static FileSystem fs = null;
+    private static Path pt;
+
+    @BeforeClass
+    static public void setUp() throws IOException {
+        Configuration config = new Configuration();
+//adding local hadoop configuration
+        config.addResource(new Path("/usr/local/Cellar/hadoop/2.4.0/libexec/etc/hadoop/core-site.xml"));
+        config.addResource(new Path("/usr/local/Cellar/hadoop/2.4.0/libexec/etc/hadoop/hdfs-site.xml"));
```
Comment (Hexiaoqiao) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java]: please change this configuration file to relative path.

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/ViewFsCarbonFileTest.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+                return fileStatus;
+            }
+
+        };
+        viewFSCarbonFile = new ViewFSCarbonFile(fileStatus);
+        assertTrue(viewFSCarbonFile.listFiles(carbonFileFilter).length == 0);
+    }
+
+    @Test
+    public void testParentFIle() {
```
Comment (Hexiaoqiao) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/ViewFsCarbonFileTest.java]: testParentFile?

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                        public void progress() {
+                            LOGGER.info("Started Writing to File===");
+                        }
+                    });
+            BufferedWriter br = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+            br.write("Hello World");
+            br.close();
+            fs.close();
+
+            fileStatus = new FileStatus(12L, true, 60, 120l, 180L, new Path(fs.getName()));
```
Comment (Hexiaoqiao) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java]: why use URI (fs.getName()) to create FileStatus?

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java (hunk 1 above)
Comment (anubhav100) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java]: i have done changes can you have a look?

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+
+import static java.lang.System.out;
```
Comment (Hexiaoqiao) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java]: suggest to remove unused import.

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/AlluxioCarbonFileTest.java [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+                return fileStatus;
+            }
+
+        };
+        alluxioCarbonFile = new AlluxioCarbonFile(fileStatus);
+        assertTrue(alluxioCarbonFile.listFiles(carbonFileFilter).length == 0);
+    }
+
+    @Test
+    public void testgetParentFile() {
```
Comment (Hexiaoqiao) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/AlluxioCarbonFileTest.java]: uniform Naming for parameters: testGetParentFile?

File: core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java (hunk 4 above)
Comment (anubhav100) [inline core/src/test/java/org/apache/carbondata/core/carbon/datastorage/filesystem/HDFSCarbonFileTest.java]: @Hexiaoqiao can you have another look please

**Comment by Hexiaoqiao** [comment]
@anubhav100  there are still some exception when run ut, pls ref: http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/616/consoleFull

**Comment by anubhav100** [comment]
@Hexiaoqiao thats because we are writing the test cases for exception take a look here 

@Override public String getCanonicalPath() {
    try {
      return file.getCanonicalPath();
    } catch (IOException e) {
      LOGGER
          .error(e, "Exception occured" + e.getMessage());
    }
    return null;
  }
Logger.error is printing those exception

**Comment by Hexiaoqiao** [comment]
LGTM
@anubhav100  thanks for ur contributions. pls rename title following format as "[CARBONDATA-<Jira issue #>] Description of pull request".

**Comment by anubhav100** [comment]
@Hexiaoqiao i renamed pull request

**Comment by chenliang613** [comment]
@anubhav100 the below info only be template for reminding to check everything,  the PR description doesn't need to include , please remove it.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 Make sure the PR title is formatted like:
[CARBONDATA-<Jira issue #>] Description of pull request

 Make sure tests pass via mvn clean verify. (Even better, enable
Travis-CI on your fork and ensure the whole test matrix passes).

 Replace <Jira issue #> in the title with the actual Jira issue
number, if there is one.

 If this contribution is large, please file an Apache
Individual Contributor License Agreement.

 Testing done

 Please provide details on 
 - Whether new unit test cases have been added or why no new tests are required?
 - What manual testing you have done?
 - Any additional information to help reviewers in testing this change.
 For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

**Comment by anubhav100** [comment]
@chenliang613 i removed template from pr thanks

**Comment by anubhav100** [comment]
@chenliang613 i removed template from pr please merge it

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**